### PR TITLE
Sample Stitch push notification app

### PR DESCRIPTION
**Pull request by kay-kim**
@edaniels / @jsflax  -- The example is slightly doofusy but could you all take a look at the MainActivity.java?  The example works and all, but wasn't sure whether the way I do stuff is normal in android app development.

### CODE REVIEW

**Comment by edaniels** [comment]
Jason is on vacation but I want to allow him to review this. Cool if we wait?

**Comment by kay-kim** [comment]
Yup.  Can wait

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    private static final String TAG = "StitchPushNotification";
+
+    private StitchClient stitchClient;
+
+    private GCMPushClient pushClient;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        stitchClient =  new StitchClient(this, "STICH-APP-ID");
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: extra space after `=`

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        initLogin();
+
+        setContentView(R.layout.activity_main);
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: any reason you're calling `setContentView` twice here?

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                                    initGCMClient();
+                                } else {
+                                    Log.e(TAG, "Error logging in anonymously", task.getException());
+                                }
+                            }
+                        });
+                    } else {
+                        Log.e(TAG, "Enable Anonymous Login", task.getException());
+                    }
+
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: extra line

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+public class MainActivity extends AppCompatActivity {
+
+    public static final String TOPIC_HOLIDAYS = "com.mongodb.example.pushnotification.holidays";
+    public static final String TOPIC_QUOTES = "com.mongodb.example.pushnotification.quotes";
+
+    private static final String TAG = "StitchPushNotification";
+
+    private StitchClient stitchClient;
+
+    private GCMPushClient pushClient;
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: needs new line

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                });
+            }
+        });
+    }
+
+
+    /** Called when the user registers for push notification topics
+     * @param view
+     */
+    public void subscribeToTopic(View view) {
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: Should add a comment here noting that this button is the `onClick` method for `button2`. Setting `onClick` in the XML, while convenient (and totally fine), is a bit of an antipattern, so it's always nice to give the developer some breadcrumbs in the actual Java file as to why this method exists.

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        }
+
+        final TextView txtView = (TextView) findViewById(R.id.textView);
+        txtView.setText("");
+        txtView.setVisibility(View.INVISIBLE);
+        String subscriptionStatus = "";
+
+        CheckBox chkboxHolidays = (CheckBox) findViewById(R.id.checkBox_holidays);
+
+        if (chkboxHolidays.isChecked()){
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: extra space needed after last paren

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                @Override
+                public void onComplete(@NonNull final Task<Void> task) {
+                    if (!task.isSuccessful()) {
+                        Log.d(TAG, "Error subscribing to topic " + task.getException());
+                        return;
+                    }
+                    Log.d(TAG, "Subscribed to topic Holidays");
+                    txtView.append("Subscribed to topic Holidays.\n");
+                }
+
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: extra new line

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+                        Log.d(TAG, "Error Unsubscribing to topic " + task.getException());
+                        return;
+                    }
+                    Log.d(TAG, "Successfully Unsubscribed to topic Quotes.");
+
+                }
+
+            });
+        }
+        txtView.setVisibility(View.VISIBLE);
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: [q] What was the point of initially setting this to `INVISIBLE`?

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+            });
+        } else {
+            pushClient.unsubscribeFromTopic(TOPIC_HOLIDAYS).addOnCompleteListener(new OnCompleteListener<Void>() {
+                @Override
+                public void onComplete(@NonNull final Task<Void> task) {
+                    if (!task.isSuccessful()) {
+                        Log.d(TAG, "Error Unsubscribing to topic " + task.getException());
+                        return;
+                    }
+                    Log.d(TAG, "Successfully Unsubscribed to topic Holidays");
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: should have new line above and no new line below

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+
+        if (chkboxQuotes.isChecked()){
+            pushClient.subscribeToTopic(TOPIC_QUOTES).addOnCompleteListener(new OnCompleteListener<Void>() {
+                @Override
+                public void onComplete(@NonNull final Task<Void> task) {
+                    if (!task.isSuccessful()) {
+                        Log.d(TAG, "Error subscribing to topic " + task.getException());
+                        return;
+                    }
+                    Log.d(TAG, "Subscribed to topic Quotes.");
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: nit: should have new line above

File: StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+                    txtView.append("Subscribed to topic Quotes.");
```
Comment (jsflax) [inline StitchSamplePushNotificationApp/app/src/main/java/com/mongodb/example/pushnotification/MainActivity.java]: [q] Why does this `append` not also append a new line?

**Comment by jsflax** [comment]
LGTM!

**Pull request merged**